### ISSUE TITLE
What is the MAX value "height" can take?

### ISSUE THREAD

**Issue by Shivani912** [informalsystems/tendermint-rs#830]
Height is a `u64` which means it can take values up to `u64::MAX` but the below implementation of `TryFrom` seems to allow values up to only `i64::MAX`
https://github.com/informalsystems/tendermint-rs/blob/ebc3ac9ac0a4934b77814679f05a0b7a3c7c26b3/tendermint/src/block/height.rs#L34-L40

Curious to know why is it so?

**Comment by andrey-kuprianov** [comment]
The answer is in [Tendermint-go](https://github.com/tendermint/tendermint/blob/5c547137f627e2728729b4147e4afe397aba27a5/types/block.go#L337).

So it is for compatibility, and the check is not redundant.

**Comment by thanethomson** [comment]
`Height` is technically a "domain type", which implies that it's only used internally within `tendermint-rs`. From what I understand this allows us to enforce non-negative heights using the type system.

When communicating externally, as @andrey-kuprianov mentioned, we always using signed 64-bit integers, which implies that the maximum value a height can realistically take is `i64::MAX` (see https://doc.rust-lang.org/std/primitive.i64.html#associatedconstant.MAX).

**Comment by Shivani912** [comment]
Thanks for explaining @thanethomson ! There is this [PR](https://github.com/tendermint/tendermint/pull/6025) on tendermint-go for migration of `Height` from `int64` to `uint64` and also the [spec](https://github.com/tendermint/spec/blob/master/spec/core/data_structures.md#header) says `Height` is supposed to be `uint64`. So I guess this will bring the change on tendermint-rs too, right?

I'll keep this issue open as a means to track this situation :)

**Comment by thanethomson** [comment]
> There is this [PR](https://github.com/tendermint/tendermint/pull/6025) on tendermint-go for migration of `Height` from `int64` to `uint64` and also the [spec](https://github.com/tendermint/spec/blob/master/spec/core/data_structures.md#header) says `Height` is supposed to be `uint64`. So I guess this will bring the change on tendermint-rs too, right?

Yes, definitely, it'll simplify things quite a bit. And that's good news it's being migrated to `uint64` :+1:
